docs/machine-learning/how-to-guides/prepare-data-ml-net.md

```diff
@@ -0,0 +1,331 @@
+
+ML.NET supports the following text transforms:
+
+- [ApplyWordEmbedding](xref:Microsoft.ML.TextCatalog.ApplyWordEmbedding*)
+- [FeaturizeText](xref:Microsoft.ML.TextCatalog.FeaturizeText*)
+- [LatentDirichletAllocation](xref:Microsoft.ML.TextCatalog.LatentDirichletAllocation*)
+- [NormalizeText](xref:Microsoft.ML.TextCatalog.NormalizeText*)
+- [ProduceHashedNGrams](xref:Microsoft.ML.TextCatalog.ProduceHashedNgrams*)
+- [ProduceHashedWordBags](xref:Microsoft.ML.TextCatalog.ProduceHashedWordBags*)
+- [ProduceNGrams](xref:Microsoft.ML.TextCatalog.ProduceNgrams*)
+- [ProduceWordBags](xref:Microsoft.ML.TextCatalog.ProduceWordBags*)
+- [RemoveDefaultStopWords](xref:Microsoft.ML.TextCatalog.RemoveDefaultStopWords*)
+- [RemoveStopWords](xref:Microsoft.ML.TextCatalog.RemoveStopWords*)
+- [TokenizeIntoCharactersAsKeys](xref:Microsoft.ML.TextCatalog.TokenizeIntoCharactersAsKeys*)
+- [TokenizeIntoWords](xref:Microsoft.ML.TextCatalog.TokenizeIntoWords*)
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+ReviewData[] reviews = new ReviewData[]
+{
+    new ReviewData
+    {
+        Description="This is a good product",
+        Rating=4.7f
+    },
+    new ReviewData
+    {
+        Description="This is a bad product",
+        Rating=2.3f
+    }
+};
+```
+
+The minimum step to convert text to a numerical vector representation is to use the [`FeaturizeText`](xref:Microsoft.ML.TextCatalog.FeaturizeText*) method.
+
+```csharp
+// Define text transform estimator
+var textEstimator  = mlContext.Transforms.Text.FeaturizeText("Description");
+
+// Build ITransformer that performs operation
+ITransformer textTransformer = textEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = textTransformer.Transform(data);
+```
+
+The resulting transform would convert the text values in the `Description` column to a numerical vector that looks similar to the output below:
+
+```text
+[ 0.2041241, 0.2041241, 0.2041241, 0.4082483, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0.2041241, 0, 0, 0, 0, 0.4472136, 0.4472136, 0.4472136, 0.4472136, 0.4472136, 0 ]
+```
+
+More complex text processing steps can be combined to remove noise and potentially reduce the amount of resources required for processing.
+
+```csharp
+// Define text transform estimator
+var textEstimator = mlContext.Transforms.Text.NormalizeText("Description")
+    .Append(mlContext.Transforms.Text.TokenizeIntoWords("Description"))
+    .Append(mlContext.Transforms.Text.RemoveDefaultStopWords("Description"))
+    .Append(mlContext.Transforms.Text.FeaturizeText("Description"));
+```
+
+Using the first entry as an example, below is a detailed description of the results produced by the transformation steps defined in the code sample above:
```

Comment:
"below" -> "the following"
"defined in the code sample above" -> "defined in the previous code sample"